c++/cpp2py/converters/variant.hpp

```diff
@@ -45,18 +45,14 @@ namespace cpp2py {
     }
 
     struct _visitor {
-      template <typename U> PyObject *operator()(U const &x) { return py_converter<U>::c2py(x); }
+      template <typename U> PyObject *operator()(U x) { return py_converter<std::decay_t<U>>::c2py(std::move(x)); }
```

Comment:
We could use a lambda, and 
convert(..) function to avoid the py_converter explicit call...
No visitor struct really needed..

Reply:
Sure this can be done. Thats more a matter of taste though? Not really less code.

Reply:
Lambda will be more concise with c++20 templated lambdas, so we can put that in the future.

Reply:
yes, it is a question of style. Agreed for C++20